test/state/state.cpp

```diff
@@ -415,48 +420,32 @@ StateDiff finalize(const StateView& state_view, evmc_revision rev, const address
     return state.build_diff(rev);
 }
 
-std::variant<TransactionReceipt, std::error_code> transition(const StateView& state_view,
-    const BlockInfo& block, const BlockHashes& block_hashes, const Transaction& tx,
-    evmc_revision rev, evmc::VM& vm, int64_t block_gas_left, int64_t blob_gas_left)
+TransactionReceipt transition(const StateView& state_view, const BlockInfo& block,
+    const BlockHashes& block_hashes, const Transaction& tx, evmc_revision rev, evmc::VM& vm,
+    int64_t execution_gas_limit)
 {
     State state{state_view};
-    auto* sender_ptr = state.find(tx.sender);
-
-    // Validate transaction. The validation needs the sender account, so in case
-    // it doesn't exist provide an empty one. The account isn't created in the state
-    // to prevent the state modification in case the transaction is invalid.
-    const auto validation_result =
-        validate_transaction((sender_ptr != nullptr) ? *sender_ptr : Account{}, block, tx, rev,
-            block_gas_left, blob_gas_left);
-
-    if (holds_alternative<std::error_code>(validation_result))
-        return get<std::error_code>(validation_result);
 
-    // Once the transaction is valid, create new sender account.
-    // The account won't be empty because its nonce will be bumped.
-    auto& sender_acc = (sender_ptr != nullptr) ? *sender_ptr : state.insert(tx.sender);
-
-    assert(sender_acc.nonce < Account::NonceMax);  // Checked in transaction validation
+    auto& sender_acc = state.get_or_insert(tx.sender);
+    assert(sender_acc.nonce < Account::NonceMax);  // Required for valid tx.
     ++sender_acc.nonce;                            // Bump sender nonce.
 
-    const auto execution_gas_limit = get<int64_t>(validation_result);
-
     const auto base_fee = (rev >= EVMC_LONDON) ? block.base_fee : 0;
-    assert(tx.max_gas_price >= base_fee);                   // Checked at the front.
-    assert(tx.max_gas_price >= tx.max_priority_gas_price);  // Checked at the front.
+    assert(tx.max_gas_price >= base_fee);                   // Required for valid tx.
+    assert(tx.max_gas_price >= tx.max_priority_gas_price);  // Required for valid tx.
     const auto priority_gas_price =
         std::min(tx.max_priority_gas_price, tx.max_gas_price - base_fee);
     const auto effective_gas_price = base_fee + priority_gas_price;
 
```

Comment:
I would remove the empty line and and add one after assert.

Reply:
:point_up: